### DISCUSSION TITLE
Generate metrics for test data with train.py script

### DISCUSSION THREAD

**Discussion by pasmai** [ultralytics/yolov5#6209] [Q&A]
I divided the data in my .yaml file into the 3 categories:

```
train:  train/images
val:    valid/images
test:   test/images
```

Using `train.py` with the Weights & Bias overview I can only see charts for training and validation. I would like to see what the precision/recall/loss on the test dataset is.
To figure out if I run into any issues with overfitting or how many epochs make sense to run.

So essentially these graphs run with the same model but on the test data instead of the training:
![image](https://user-images.githubusercontent.com/14838120/148217250-f0bdeaad-f6b6-4702-90a6-3a4cf28d96b9.png)

Can anybody help me out on how to achieve this?
Cheers

**Comment by glenn-jocher** [comment]
@pasmai validation set is used during training. Test set is not supposed to be used during training, only after training is complete for an independent evaluation of your metrics. You can evaluate a trained model on your test set with `val.py --task` test:

```bash
python val.py --weights path/to/best.pt --data path/to/data.yaml --task test
```

**Reply by pasmai** [reply]
Ah thanks you for your reply.

Ignore. I found [this explenation](https://blog.roboflow.com/train-test-split/) that helped.
~~Just to make sure: Does the result of the validate set feed back into training (to influence the weights)? Or does it only use those for validation during the testing to track the progress of accuracy and possible overfitting?~~
